src/coreclr/src/pal/src/init/pal.cpp

```diff
@@ -1265,6 +1269,139 @@ static LPWSTR INIT_FormatCommandLine (int argc, const char * const *argv)
     return retval;
 }
 
+#if defined(__linux__)
+#define symlinkEntrypointExecutable "/proc/self/exe"
+#elif !defined(__APPLE__)
+#define symlinkEntrypointExecutable "/proc/curproc/exe"
+#endif
+
+bool GetAbsolutePath(const char* path, PathCharString& absolutePath)
```

Comment:
We now have this (very similar) logic replicated in three places in the repo. Other two are:

1. https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/coreclr/src/hosts/unixcoreruncommon/coreruncommon.cpp#L53-L67
2. https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/installer/corehost/cli/hostmisc/pal.unix.cpp#L729-L812

it would be nice to de-duplicate some of it. 🙂

Reply:
(other two support FreeBSD and SunOS)

Reply:
@am11 yeah, I am not happy with that. I was trying to figure out a reasonable way to do that, but everything I could come up with was messy. corerun needs this stuff before the coreclr with PAL is loaded, so it cannot call it from the PAL. All the other apps like ildasm, ilasm etc. are built on top of the PAL, so they cannot include platform headers. And I didn't want to modify all these by adding a call to the exe extraction path. I was also thinking that maybe I can relax the coreclr_initialize requirement and not require it to pass in the absolute path to the exe and have it just in the PAL. But then I've realized that corerun needs that path for other purposes.
So I've decided to make it the way I did, since the window for 5.0 changes without having to go through extra process closes this week.
I can see a possible semi-unification in a future though - I'd like to see the corehost native hosts to be built in the coreclr, so at least the corerun and the dotnet host could reasonably share that code.

Reply:
> unification in a future 

I think we need to adopt mini-pals like what is under https://github.com/dotnet/runtime/tree/master/src/libraries/Native/Unix/Common more broadly.

Reply:
Would it make sense (in the future) to extract some functions in a common code file and include in the individual executable projects?